### ISSUE TITLE
Kubernetes connection should refresh max 3 times then requires user input

### ISSUE THREAD

**Issue by axel7083** [containers/podman-desktop#8394]
### Is your enhancement related to a problem? Please describe

We discussed this during the daily of 8 august 2024 where we discussed the way we handle Kubernetes informers (aka connections).

First we have a logic, of checking the number of contexts in the kubeconfig, if we have less than 10, we try to create informers _for each of them_: IMO this is probably not something the user is expecting (might be ?).

https://github.com/containers/podman-desktop/blob/79ebd8678a868777780cde3f1f4f3f87a05b46ba/packages/main/src/plugin/kubernetes-context-state.ts#L69-L71

We have a backoff system to restart/reconnec the informers:

https://github.com/containers/podman-desktop/blob/79ebd8678a868777780cde3f1f4f3f87a05b46ba/packages/main/src/plugin/kubernetes-context-state.ts#L924

- backoffInitialValue: initial delay between two connection attempts (1000)
- backoffLimit: maximum delay between two connection attempts (60_000)

So if we have 9 contexts, we are tyring to reconnect to each one of them every 60 seconds, this forever (from my understanding).

### Describe the solution you'd like

Let's look at how `OpenLens` is handling the reconnection scenario. When trying to connect to a cluster, we are getting an error directly on the page.

![image](https://github.com/user-attachments/assets/dd63fb4e-476c-4b7b-8167-b8f2eb227220)

OpenLens does not seems to create the informers before the cluster can be reached. It seems to uses the client to check for the version beforehand (from my understanding.)

We are also showing the connection status information, but we lack of user interaction to allow direct re-connection.

![image](https://github.com/user-attachments/assets/020581f1-cc31-4037-b17b-5c5e9a5b0367)

![image](https://github.com/user-attachments/assets/1d130f77-a151-4ef9-8422-ba02701f61cb)

### Describe alternatives you've considered

There is a discussion on the refresh actions, but this issue is focused on improving the connection pool system.

**Comment by axel7083** [comment]
See https://github.com/containers/podman-desktop/issues/8341 for follow up

**Comment by axel7083** [comment]
cc @deboer-tim @cdrage @feloy

**Comment by deboer-tim** [comment]
Agreed as per the discussion, instead of backing off and continuing to ping very infrequently (and indefinitely), we should have some backoff / several tries and give up, with a user action to try again.

Why? In practise I almost always know when the connection state has changed (e.g. provider started or network fixed) and want to try again now, not wait up to 60s. No point using resource to poll multiple contexts indefinitely if I need a manual action anyway.

We could probably add the action first and then remove the indefinite pings?

We should track any discussion about the 10 limit separately, and maybe we want different behaviour in the Settings page (e.g. we try every context once). I also think we should take provider state into account and opened #8400.

**Comment by feloy** [comment]
> OpenLens does not seems to create the informers before the cluster can be reached. It seems to uses the client to check for the version beforehand (from my understanding.)

I have made this proposal, to have a similar result, with the advantage of getting permissions to access the different resources, and run the appropriate informers only: https://github.com/containers/podman-desktop/issues/7629

**Comment by axel7083** [comment]
I will plan a meeting to discuss this mater more in depth

**Comment by feloy** [comment]
After discussion on today's team meeting (11/07), we will check the connectivity using the cluster health check endpoint, and we will stop after a few tries. There is already a button for the user to ask for retrying.

**Comment by feloy** [comment]
setting on hold, depends on https://github.com/podman-desktop/podman-desktop/issues/10467

**Comment by feloy** [comment]
In Kubernetes experimental mode, the check is done only once, with an health check. If the check fails, there is no retry, and the user has 'Connect' buttons available to retry the connection.

@axel7083 is this fixing this issue?

**Comment by deboer-tim** [comment]
Personally I'd prefer the retry just to rule out timing, busy machine, or random failure, but I don't feel strongly about it / am willing to give it a try.

**Comment by feloy** [comment]
There is a 5s timeout in the health check, which may be increased in case we want to support timeouts.

I propose to close this issue as Not planned, as I cannot see advantage of retrying the health check in the current (experimental) implementation. In case of random failure, the user can still manually retry the connection.

Please re-open the issue if you think there is any advantage
